prow/pjutil/abort.go

```diff
@@ -36,35 +37,79 @@ type prowClient interface {
 // message.
 type ProwJobResourcesCleanup func(pj prowapi.ProwJob) error
 
+func refsForRefs(ref prowapi.Refs) refs {
+	var pulls []int
+	for _, pull := range ref.Pulls {
+		pulls = append(pulls, pull.Number)
+	}
+	return refs{
+		org:   ref.Org,
+		repo:  ref.Repo,
+		base:  ref.BaseRef,
+		pulls: pulls,
+	}
+}
+
 // jobIndentifier keeps the information required to uniquely identify a prow job
 type jobIndentifier struct {
```

Comment:
We may want to also include the job type in the identifier so that if a job is run as both a presubmit and as a batch of the same refs we don't consider them duplicates. FWIW this shouldn't happen today, since batches always have more than 1 PR and presubmits always have exactly 1, but we don't have anything that enforces that so it might be nice to handle.
/shrug